src/Illuminate/Support/Collection.php

```diff
@@ -201,9 +201,19 @@ public function first(callable $callback = null, $default = null)
      *
      * @return static
      */
-    public function flatten()
+    public function flatten($depth = INF)
```

Comment:
Also, you forgot to add phpdoc. If you keep INF, the phpdoc needs to be `int|float` since INF is actually a float, according to var_dump.